### ISSUE TITLE
[BUG] b64decode does not handle whitespaces

### ISSUE THREAD

**Issue by martinvuyk** [modularml/mojo#3446]
### Bug description

Detected by @lemire in PR #3443

> Currently, b64decode does not appear to handle white-space characters. I would have expected the following to print 'Bonjour', it does not:

```mojo
from base64 import b64decode
def main():
    var data = b64decode("Qm9 uam91cg==")
    print(data)
```
output:
```text
Bo������
```

### Steps to reproduce

- Include relevant code snippet or link to code that did not work as expected.
- If applicable, add screenshots to help explain the problem.
- If using the Playground, name the pre-existing notebook that failed and the steps that led to failure.
- Include anything else that might help us debug the issue.

### System information

```shell
- What OS did you do install Mojo on ?
- Provide version information for Mojo by pasting the output of `mojo -v`
`mojo 2024.9.105`
- Provide Modular CLI version by pasting the output of `modular -v`
```

**Comment by lemire** [comment]
Note that I did not report it as a bug because the documentation does not seem to imply that white space is handled.

A relevant specification is WHATWG Forgiving Base64 decoding:

https://infra.spec.whatwg.org/#forgiving-base64-decode

C#/.NET follows it, as well as the JavaScript's `atob` function. Possibly other systems follow it as well.

**Comment by lemire** [comment]
@martinvuyk Right. So the base64 algorithm in [simdutf](https://github.com/simdutf/simdutf) can solve this at high speed. It is already used in production. (It is part of WebKit/Safari and Node.js, Bun, etc.)
